lib/puppet/functions/dnsquery/rlookup.rb

```diff
@@ -1,36 +1,31 @@
 # frozen_string_literal: true
 
 # Retrieves results from DNS reverse lookup and returns it as an array.
-# Each record in the array will be a hostname.
-# An optional lambda can be given to return a default value in case the
-# lookup fails. The lambda will only be called if the lookup failed.
 Puppet::Functions.create_function(:'dnsquery::rlookup') do
+  # @param address the ip address to lookup
+  # @param block an optional lambda to return a default value in case the lookup fails
+  # @return An array of hostnames matching the ip address
   dispatch :dns_rlookup do
     param 'String', :address
-  end
-
-  dispatch :dns_rlookup_with_default do
-    param 'String', :address
-    block_param
+    optional_block_param :block
+    return_type 'Array[String]'
   end
 
   def dns_rlookup(address)
     addr = IPAddr.new(address)
-    Resolv::DNS.new.getresources(
+    ret = Resolv::DNS.new.getresources(
       addr.reverse, Resolv::DNS::Resource::IN::PTR
     ).map do |res|
       res.name.to_s
     end
-  end
+    return ret unless block_given?
 
-  def dns_rlookup_with_default(address)
-    ret = dns_rlookup(address)
     if ret.empty?
       yield
     else
       ret
     end
```

Comment:
Would `block_given? && ret.empty? ? yield : ret` be better than to have a separate return + if/else statement? Or am I code golfing too much?

Reply:
no much better thanks